sched/sched/sched.h

```diff
@@ -56,6 +56,8 @@
 #  define this_task()            (current_task(this_cpu()))
 #endif
 
+#define is_idle_task(t)          ((t)->pid == this_cpu())
```

Comment:
Why we compare against `this_cpu()` and not against `IDLE_PROCESS_ID`. Does this mean that we have idle task per core in SMP mode?